diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6aa0fb8..cc7933f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -417,6 +417,9 @@ if(NOT ANDROID)
 
 		# Note: We do not need to explicitly check for X11 as it is done in the cmake
 		# FindOpenGL module on linux.
+		if(USE_X11 AND NOT X11_FOUND)
+			pkg_check_modules(X11 x11)
+		endif()
 		if(USE_X11 AND X11_FOUND)
 			set(USE_X11 1)
 			add_definitions(-DHAVE_X11=1)
